### PR TITLE
cds.test with authentication

### PR DESCRIPTION
**Pull request by larslutz96**
Our documentation nicely explains how to use service APIs and HTTP APIs with cds.test.

We should add a link to the authentication configuration and explain which kind is used for cds.test. In addition, we should add samples how to set a user for the above mentioned APIs.

### CODE REVIEW

File: node.js/cds-test.md [hunk 1]
```diff
@@ -82,6 +82,23 @@ await POST (`/browse/submitOrder`, { book: 201, quantity: 5 })
 [Learn more in GET/PUT/POST.](#http-bound) {.learn-more}
 
 
+#### Authenticated Endpoints
+`cds.test()` uses [mocked authentication](https://pages.github.tools.sap/cap/docs/node.js/authentication#mocked). To configure authentication, you can utilize `cds.requires.auth` in the `package.json` file or directly in your test file like so:
```
Comment (chgeo) [inline node.js/cds-test.md:86]: It's a bit misleading as `cds.test` itself doesn't change/set the auth mode.  So rather:

```suggestion
`cds.test()` uses the standard authentication strategy in development mode, which is the [mocked authentication](../node.js/authentication#mocked). To configure it, use `cds.requires.auth` in the `package.json` file or directly in your test file like so:
```

File: node.js/cds-test.md [hunk 2]
```diff
@@ -82,6 +82,23 @@ await POST (`/browse/submitOrder`, { book: 201, quantity: 5 })
...
+
+```js
+cds.requires.auth.users.foo = {
```
Comment (chgeo) [inline node.js/cds-test.md:89]: Should be `cds.env`:

```suggestion
cds.env.requires.auth.users.foo = {
```

**Review by chgeo — COMMENTED** [review]
Nice.  See comments.

**Comment by renejeglinsky** [comment]
Closes: #371

File: node.js/cds-test.md [hunk 3]
```diff
@@ -82,6 +82,23 @@ await POST (`/browse/submitOrder`, { book: 201, quantity: 5 })
 [Learn more in GET/PUT/POST.](#http-bound) {.learn-more}
 
 
+#### Authenticated Endpoints
+`cds.test()` uses the standard authentication strategy in development mode, which is the [mocked authentication](../node.js/authentication#mocked). To configure it, use `cds.requires.auth` in the `package.json` file or directly in your test file like so:
```
Comment (renejeglinsky) [inline node.js/cds-test.md:86]:
```suggestion
`cds.test()` uses the standard authentication strategy in development mode, which is the [mocked authentication](../node.js/authentication#mocked). To configure it, use `cds.requires.auth` in the _package.json_ file or directly in your test file as follows:
```

**Review by renejeglinsky — APPROVED** [review]

**Review by chgeo — APPROVED** [review]

File: node.js/cds-test.md [hunk 4]
```diff
@@ -82,6 +82,23 @@ await POST (`/browse/submitOrder`, { book: 201, quantity: 5 })
 [Learn more in GET/PUT/POST.](#http-bound) {.learn-more}
 
 
+#### Authenticated Endpoints
+`cds.test()` uses the standard authentication strategy in development mode, which is the [mocked authentication](../node.js/authentication#mocked). To configure it, use `cds.requires.auth` in the _package.json_ file or directly in your test file as follows:
+
+```js
```
Comment (johannes-vogel) [inline node.js/cds-test.md:88]: I think the recommended way is to do that in package.json:

```jsonc
{
  "requires": { 
    "auth": { 
      "[development]": {
        "users":  { 
          "foo": {
            "password": "bar",
            "roles": { "admin": true, "cds.Subscriber": true },
            "attr": { "name": "Foo" }
} } } } } }
```

Comment (larslutz96) [reply node.js/cds-test.md:88]: @johannes-vogel So we should remove the other option or make it clear that the package.json is the recommended option?

Comment (johannes-vogel) [reply node.js/cds-test.md:88]: I'd leave out the code variant. That's error prone and might have side effects.

**Review by johannes-vogel — APPROVED** [review]

File: node.js/cds-test.md [hunk 5]
```diff
@@ -83,15 +83,7 @@ await POST (`/browse/submitOrder`, { book: 201, quantity: 5 })
...
-`cds.test()` uses the standard authentication strategy in development mode, which is the [mocked authentication](../node.js/authentication#mocked). To configure it, use `cds.requires.auth` in the _package.json_ file or directly in your test file as follows:
-
-```js
-cds.env.requires.auth.users.foo = {
-  password: 'bar',
-  roles: { admin: true, ['cds.Subscriber']: true },
-  attr: { name: 'Foo' }
-}
-```
+`cds.test()` uses the standard authentication strategy in development mode, which is the [mocked authentication](../node.js/authentication#mocked). To configure it, use `cds.requires.auth` in the _package.json_ file as described in the documenation for the mocked authentication.
```
Comment (renejeglinsky) [inline node.js/cds-test.md]: If mocked auth is used, can the pre-configured mock users, like alice, be used? If so, we should adapt the snippet in line 99.

Comment (johannes-vogel) [reply node.js/cds-test.md]: yes they can be used

File: node.js/cds-test.md [hunk 6]
```diff
@@ -82,6 +82,16 @@ await POST (`/browse/submitOrder`, { book: 201, quantity: 5 })
...
+#### Authenticated Endpoints
+`cds.test()` uses the standard authentication strategy in development mode, which is the [mocked authentication](../node.js/authentication#mocked). This also includes the usage of [pre-definded mock users](../node.js/authentication#mock-users)
+
+You can set the user for an authenticated request like this:
+
+```js
+await GET('/admin/Books', { auth: { username: 'alice', password: '' } })
+```
+
+[Learn how to explicitly configure mock user in your _package.json_ file.](../node.js/authentication#mocked){.learn-more}
```
Comment (renejeglinsky) [inline node.js/cds-test.md:94]:
```suggestion
[Learn how to explicitly configure mock users in your _package.json_ file.](../node.js/authentication#mocked){.learn-more}
```

**Review by renejeglinsky — APPROVED** [review]
Please check if my last edit is fine for both of you.

**Comment by larslutz96** [comment]
> Please check if my last edit is fine for both of you.

@johannes-vogel seems good to me

**Pull request merged**